src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerGenotypingEngine.java

```diff
@@ -634,8 +628,15 @@ protected static void cleanUpSymbolicUnassembledEvents( final List<Haplotype> ha
         haplotypes.removeAll(haplotypesToRemove);
     }
 
-    protected static Map<Allele, List<Haplotype>> createAlleleMapper(final Map<VariantContext, Allele> mergeMap, final Map<Event, List<Haplotype>> eventMap ) {
-        return mergeMap.entrySet().stream().collect(Collectors.toMap(e -> e.getValue(), e -> eventMap.get(new Event(e.getKey())), (x,y)->x, LinkedHashMap<Allele, List<Haplotype>>::new));
+
+    protected static Map<Allele, List<Haplotype>> createAlleleMapper(List<VariantContext> eventsAtThisLoc, Map<Event, List<Haplotype>> eventMapper, VariantContext mergedVC) {
```

Comment:
Got it. I always wondered why `private static` methods were ever helpful but now I understand.